starknet_py/tests/e2e/integration_tests/client_test.py

```diff
@@ -414,3 +416,45 @@ async def test_get_block(full_node_client_integration):
 
     for tx in res.transactions:
         assert tx.hash is not None
+
+
+@pytest.mark.skipif(
+    condition="--client=full_node" in sys.argv,
+    reason="Separate FullNode tests from Gateway ones.",
+)
+@pytest.mark.asyncio
+async def test_get_public_key(gateway_client_integration):
+    public_key = await gateway_client_integration.get_public_key()
+
+    assert isinstance(public_key, str)
```

Comment:
Could this be tested some better way than just checking the type? should this public_key be equal to some value we can check?

Reply:
We can, but it would have to be changed every year or in case it was leaked, wdyt?

Reply:
what's supposed to be leaked? the public key?